tools/esdlcmd/README.md

```diff
@@ -0,0 +1,424 @@
+    manifest              Build a service binding or bundle from a manifest file.
+    bind-log-transform    Configure log transform associated with existing ESDL binding.
+    unbind-log-transform  Remove log transform associated with existing ESDL binding.
+
+The sections below cover the commands in more detail.
+
+## manifest
+
+The `manifest` command creates an xml binding file that configures features of a DESDL service. It takes a manifest file as input which dictates what will be included in the output- a *binding* or a *bundle*. A *binding* has dynamic service configuration, while the *bundle* is a wrapper around the binding that adds the ESDL interface definition.
+
+How you want to configure and run your ESP determines if you'll need a binding or a bundle. When using a dali to hold dynamic configuration elements like ESDL Definitions and Service Bindings you'll need to create a *binding* for your service. This is the only option when deploying an ESP on bare metal using init.d or systemd startup scripts.
+
+Alternately, if you direct the ESP to read its dynamic configuration from files, then you'll need to create a *bundle*. This option is available when you run as an *esp application*. The esp application mode was created for cloud deployments but is also suitable for bare metal. Note that even when running in application mode, an ESP still requires a dali even if ESDL configurations aren't stored there. This dependency will be removed in a future release.
+
+### Manifest File
+
+A simplified example showing the format of a manifest file is below.
+
+```xml
+<em:Manifest xmlns:em="urn:hpcc:esdl:manifest">
+    <em:ServiceBinding service="WsFoobar" id="WsFoobar_desdl_binding">
+        <Methods>
+            <Scripts>
+                <em:Include file="WsFoobar-request-prep.xml"/>
+                <em:Include file="WsFoobar-logging-prep.xml"/>
+            </Scripts>
+            <Method name="FoobarSearch" url="127.0.0.1:8888">
+                <Scripts>
+                    <em:Include file="FoobarSearch-scripts.xml"/>
+                </Scripts>
+            </Method>
+        </Methods>
+        <LoggingManager>
+            <LogAgent transformSource="local" name="main-logging">
+                <LogDataXPath>
+                    <LogInfo name="PreparedData" xsl="log-prep">
+                </LogDataXPath>
+                <XSL>
+                    <Transform name="log-prep">
+                        <em:Include file="log-prep.xslt">
+                    </Transform>
+                </XSL>
+            </LogAgent>
+        </LoggingManager>
+    </em:ServiceBinding>
+    <em:ServiceDefinition>
+        <em:Include file="WsFoobar.ecm"/>
+    </em:ServiceDefinition>
+</em:Manifest>
+```
+
+The manifest nodes must use the namespace "urn:hpcc:esdl:manifest", and includes elements configuring asingle service.
+
+A `<em:ServiceBinding>` element contains the binding configuration for the service, including:
```

Comment:
No it is not comprehensive, but new elements could require an update depending on how their contents need to be treated. Any child nodes of `<em:SerivceBinding>` element are copied to output. Any descendant element named *Transform* or *Scripts* are handled specially - their output isn't a copy through, and they can have `<em:Include>` children.

I will need to adjust the code to handle other service-level attributes. I'd plan to handle them like I do now for the `service` and `id` attributes- copy them through to the appropriate output element. This would be good to chat about next week.

Only one `<em:SerivceBinding>` is allowed.

Reply:
After talking with Tony this is the change I made. All attributes from the `ServiceBinding` are copied through to the output `Binding` node. An optional `BindingDefinition` can be the child of `ServiceBinding`, and if present is output as the `Definition` node. Any attributes of `BindingDefinition` are copied through to the output `Definition` node. 

If `BindingDefinition` isn't present, or the required attributes aren't present, the tool will synthesize and output them as it currently does. 

This way you can define attributes in either the `ServiceBinding` or `BindingDefinition` node if you need to, but its not required for you to include a `BindingDefinition` in the manifest if you don't need any custom attributes on the output Definition.